flow360/version_check.py

```diff
@@ -0,0 +1,111 @@
+"""Client version check module"""
+import re
+import sys
+from enum import Enum
+from typing import List, Tuple, Union
+
+import requests
+from packaging.version import Version
+from requests import HTTPError
+
+from .cloud import http_util
+from .solver_version import Flow360Version as sv
+from .version import __version__
+
+
+class VersionSupported(Enum):
+    """
+    Enumeration representing the different states of version support.
+    """
+
+    YES = 1
+    NO = 2
+    CAN_UPGRADE = 3
+
+
+def get_supported_server_versions(app_name: str) -> List[str]:
+    """
+    Fetch a list of supported versions from the server
+
+    Args:
+        app_name (str): The name of the application
+
+    Returns:
+        List[str]: A list of supported version strings
+    """
+    try:
+        session = requests.Session()
+        http = http_util.Http(session)
+        response = http.get(f"versions?app_name={app_name}")
+    except HTTPError as error:
+        raise HTTPError(f"failed to retrieve the versions for {app_name}") from error
+
+    versions = [re.sub(r".+-", "", item["version"]) for item in response]
+
+    if len(versions) == 0:
+        raise RuntimeError(f"the supported versions should not be empty for {app_name}")
+
+    return versions
+
+
+def check_client_version(app_name: str) -> Tuple[VersionSupported, Union[str, Version]]:
+    """
+    Check the current client version against the available versions
+
+    Args:
+        app_name (str): The name of the application
+
+    Returns:
+        Tuple[VersionSupported, Union[str, Version]]: A tuple containing the version status
+        and the current or latest version
+    """
+    supported_versions = get_supported_server_versions(app_name)
+    latest_version = supported_versions[0]
+    current_version = __version__
+    if app_name == "flow360-python-client-v2":
+        current_version = Version(current_version)
+        is_supported = any(current_version == Version(v) for v in supported_versions)
+        can_upgrade = current_version < Version(latest_version)
+    else:
+        current_version = sv(current_version)
+        is_supported = any(current_version == sv(v) for v in supported_versions)
+        can_upgrade = current_version < sv(latest_version)
+
+    if not is_supported:
+        return VersionSupported.NO, current_version
+
+    if can_upgrade:
+        return VersionSupported.CAN_UPGRADE, latest_version
+
+    return VersionSupported.YES, current_version
+
+
+def client_version_get_info(app_name: str) -> None:
```

Comment:
Could you point me to the place where this function is called? I can not find it.